### ISSUE TITLE
Option for excluding mappings 

### ISSUE THREAD

**Issue by otbe** [SitePen/remap-istanbul#51]
Hi

let me explain what Im doing at the moment. I have a quite large electron app written in typescript. The only way to run tests for me in electron as a karma "browser" (at moment of writing) is to bundle all tests files. Im using webpack for this. This works great so far. Now I want to add coverage statistics to my test setup and ```remap-istanbul``` comes over the rainbow. Great tool so far. 👍 

Passing the json output of ```karma-coverage``` to ```remap-istanbul``` produces the following html output:
<img width="1436" alt="bildschirmfoto 2016-04-29 um 16 26 02" src="https://cloud.githubusercontent.com/assets/3391052/14919024/16dc26ba-0e27-11e6-9835-1957d6027d46.png">
The only entry thats interesting is ```src/utils/```. The json generated by ```karma-coverage``` contains only one entry and this is my generated test bundle.

What I want to do is: 
```remap-istanbul --exclude-mapping=~/,tests/,webpack,__root__/``` 
The generated html would only contain one entry for ```src/utils/```

Is there a way to archive this?

Thank you :)

**Comment by kitsonk** [comment]
I suspect the `remap-istanbul --exclude` option is not sufficient?

I think what you are saying is that while you get one big bundle of original coverage, that breaks out into a load of other files when remapped, of which you are only interested in the subset, and because of your build pipeline and tooling, it isn't easy to not collect the coverage information in the first place.

**Comment by otbe** [comment]
Yes your assumptions are right.. ```remap-istanbul --exclude``` is not sufficient.

I found a working solution for me:
1. create ```test.bundle.js``` with webpack
2. create coverage with ```karma-coverage``` by using ```test.bundle.js``` > ```coverage-final.json```
3. run ```remap-istanbul``` and produce ```coverage-remapped.json```
4. execute the following script:
```
var istanbul = require('istanbul');
var collector = new istanbul.Collector();
var reporter = new istanbul.Reporter();

var remappedJson = require('./../generated/coverage-remapped.json');
var keys = Object.keys(remappedJson);
var coverage = {};

for (var i = 0; i < keys.length; i++) {
  if (keys[ i ].startsWith('src/')) {
    coverage[ keys[ i ] ] = remappedJson[ keys[ i ] ];
  }
}

collector.add(coverage);

reporter.add('html');
reporter.write(collector, true, function() {});
```

**Comment by jas-chen** [comment]
@otbe thanks for your solution, I added it to my project [karma-sourcemap-writer](https://github.com/jas-chen/karma-sourcemap-writer). And it lets karma + webpack + babel work.

+1 for this feature!
